<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Introduction to msigdbr</title>

<script>// Hide empty <a> tag within highlighted CodeBlock for screen reader accessibility (see https://github.com/jgm/pandoc/issues/6352#issuecomment-626106786) -->
// v0.0.1
// Written by JooYoung Seo (jooyoung@psu.edu) and Atsushi Yasumoto on June 1st, 2020.

document.addEventListener('DOMContentLoaded', function() {
  const codeList = document.getElementsByClassName("sourceCode");
  for (var i = 0; i < codeList.length; i++) {
    var linkList = codeList[i].getElementsByTagName('a');
    for (var j = 0; j < linkList.length; j++) {
      if (linkList[j].innerHTML === "") {
        linkList[j].setAttribute('aria-hidden', 'true');
      }
    }
  }
});
</script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Introduction to msigdbr</h1>



<div id="overview" class="section level2">
<h2>Overview</h2>
<p>Pathway analysis is a common task in genomics research and there are many available R-based software tools. Depending on the tool, it may be necessary to import the pathways, translate genes to the appropriate species, convert between symbols and IDs, and format the resulting object.</p>
<p>The <code>msigdbr</code> R package provides Molecular Signatures Database (MSigDB) gene sets typically used with the Gene Set Enrichment Analysis (GSEA) software:</p>
<ul>
<li>in an R-friendly tidy/long format with one gene per row</li>
<li>for multiple frequently studied model organisms, such as mouse, rat, pig, zebrafish, fly, and yeast, in addition to the original human genes</li>
<li>as both gene symbols and NCBI/Entrez Gene IDs for better compatibility with pathway enrichment tools</li>
<li>that can be installed and loaded as a package without requiring additional external files</li>
</ul>
<p>Please be aware that the homologs were computationally predicted for distinct genes. The full pathways may not be well conserved across species.</p>
</div>
<div id="installation" class="section level2">
<h2>Installation</h2>
<p>The package can be installed from <a href="https://cran.r-project.org/package=msigdbr">CRAN</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">install.packages</span>(<span class="st">&quot;msigdbr&quot;</span>)</span></code></pre></div>
</div>
<div id="usage" class="section level2">
<h2>Usage</h2>
<p>Load package.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw">library</span>(msigdbr)</span></code></pre></div>
<p>All gene sets in the database can be retrieved without specifying a collection/category.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>all_gene_sets =<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">&quot;Mus musculus&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="kw">head</span>(all_gene_sets)</span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="co">#&gt; # A tibble: 6 x 17</span></span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="co">#&gt;   gs_cat gs_subcat gs_name entrez_gene gene_symbol human_entr… human_gene… gs_id gs_pmid</span></span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;         &lt;int&gt; &lt;chr&gt;             &lt;int&gt; &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt;  </span></span>
<span id="cb3-6"><a href="#cb3-6"></a><span class="co">#&gt; 1 C3     MIR:MIR_… AAACCA…      239273 Abcc4             10257 ABCC4       M126… &quot;&quot;     </span></span>
<span id="cb3-7"><a href="#cb3-7"></a><span class="co">#&gt; 2 C3     MIR:MIR_… AAACCA…      109359 Abraxas2          23172 ABRAXAS2    M126… &quot;&quot;     </span></span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="co">#&gt; 3 C3     MIR:MIR_… AAACCA…       60595 Actn4                81 ACTN4       M126… &quot;&quot;     </span></span>
<span id="cb3-9"><a href="#cb3-9"></a><span class="co">#&gt; 4 C3     MIR:MIR_… AAACCA…       11477 Acvr1                90 ACVR1       M126… &quot;&quot;     </span></span>
<span id="cb3-10"><a href="#cb3-10"></a><span class="co">#&gt; 5 C3     MIR:MIR_… AAACCA…       11502 Adam9              8754 ADAM9       M126… &quot;&quot;     </span></span>
<span id="cb3-11"><a href="#cb3-11"></a><span class="co">#&gt; 6 C3     MIR:MIR_… AAACCA…       23794 Adamts5           11096 ADAMTS5     M126… &quot;&quot;     </span></span>
<span id="cb3-12"><a href="#cb3-12"></a><span class="co">#&gt; # … with 8 more variables: gs_geoid &lt;chr&gt;, gs_exact_source &lt;chr&gt;, gs_url &lt;chr&gt;,</span></span>
<span id="cb3-13"><a href="#cb3-13"></a><span class="co">#&gt; #   gs_description &lt;chr&gt;, species_name &lt;chr&gt;, species_common_name &lt;chr&gt;,</span></span>
<span id="cb3-14"><a href="#cb3-14"></a><span class="co">#&gt; #   ortholog_sources &lt;chr&gt;, num_ortholog_sources &lt;dbl&gt;</span></span></code></pre></div>
<p>There is a helper function to show the available species.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="kw">msigdbr_species</span>()</span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="co">#&gt; # A tibble: 11 x 2</span></span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="co">#&gt;    species_name             species_common_name      </span></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="co">#&gt;    &lt;chr&gt;                    &lt;chr&gt;                    </span></span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="co">#&gt;  1 Bos taurus               cattle                   </span></span>
<span id="cb4-6"><a href="#cb4-6"></a><span class="co">#&gt;  2 Caenorhabditis elegans   roundworm                </span></span>
<span id="cb4-7"><a href="#cb4-7"></a><span class="co">#&gt;  3 Canis lupus familiaris   dog                      </span></span>
<span id="cb4-8"><a href="#cb4-8"></a><span class="co">#&gt;  4 Danio rerio              zebrafish                </span></span>
<span id="cb4-9"><a href="#cb4-9"></a><span class="co">#&gt;  5 Drosophila melanogaster  fruit fly                </span></span>
<span id="cb4-10"><a href="#cb4-10"></a><span class="co">#&gt;  6 Gallus gallus            chicken                  </span></span>
<span id="cb4-11"><a href="#cb4-11"></a><span class="co">#&gt;  7 Homo sapiens             human                    </span></span>
<span id="cb4-12"><a href="#cb4-12"></a><span class="co">#&gt;  8 Mus musculus             house mouse              </span></span>
<span id="cb4-13"><a href="#cb4-13"></a><span class="co">#&gt;  9 Rattus norvegicus        Norway rat               </span></span>
<span id="cb4-14"><a href="#cb4-14"></a><span class="co">#&gt; 10 Saccharomyces cerevisiae baker&#39;s or brewer&#39;s yeast</span></span>
<span id="cb4-15"><a href="#cb4-15"></a><span class="co">#&gt; 11 Sus scrofa               pig</span></span></code></pre></div>
<p>You can retrieve data for a specific collection, such as the hallmark gene sets.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>h_gene_sets =<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">&quot;Mus musculus&quot;</span>, <span class="dt">category =</span> <span class="st">&quot;H&quot;</span>)</span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="kw">head</span>(h_gene_sets)</span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="co">#&gt; # A tibble: 6 x 17</span></span>
<span id="cb5-4"><a href="#cb5-4"></a><span class="co">#&gt;   gs_cat gs_subcat gs_name entrez_gene gene_symbol human_entr… human_gene… gs_id gs_pmid</span></span>
<span id="cb5-5"><a href="#cb5-5"></a><span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;         &lt;int&gt; &lt;chr&gt;             &lt;int&gt; &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt;  </span></span>
<span id="cb5-6"><a href="#cb5-6"></a><span class="co">#&gt; 1 H      &quot;&quot;        HALLMA…       11303 Abca1                19 ABCA1       M5905 &quot;&quot;     </span></span>
<span id="cb5-7"><a href="#cb5-7"></a><span class="co">#&gt; 2 H      &quot;&quot;        HALLMA…       74610 Abcb8             11194 ABCB8       M5905 &quot;&quot;     </span></span>
<span id="cb5-8"><a href="#cb5-8"></a><span class="co">#&gt; 3 H      &quot;&quot;        HALLMA…       52538 Acaa2             10449 ACAA2       M5905 &quot;&quot;     </span></span>
<span id="cb5-9"><a href="#cb5-9"></a><span class="co">#&gt; 4 H      &quot;&quot;        HALLMA…       11363 Acadl                33 ACADL       M5905 &quot;&quot;     </span></span>
<span id="cb5-10"><a href="#cb5-10"></a><span class="co">#&gt; 5 H      &quot;&quot;        HALLMA…       11364 Acadm                34 ACADM       M5905 &quot;&quot;     </span></span>
<span id="cb5-11"><a href="#cb5-11"></a><span class="co">#&gt; 6 H      &quot;&quot;        HALLMA…       11409 Acads                35 ACADS       M5905 &quot;&quot;     </span></span>
<span id="cb5-12"><a href="#cb5-12"></a><span class="co">#&gt; # … with 8 more variables: gs_geoid &lt;chr&gt;, gs_exact_source &lt;chr&gt;, gs_url &lt;chr&gt;,</span></span>
<span id="cb5-13"><a href="#cb5-13"></a><span class="co">#&gt; #   gs_description &lt;chr&gt;, species_name &lt;chr&gt;, species_common_name &lt;chr&gt;,</span></span>
<span id="cb5-14"><a href="#cb5-14"></a><span class="co">#&gt; #   ortholog_sources &lt;chr&gt;, num_ortholog_sources &lt;dbl&gt;</span></span></code></pre></div>
<p>Retrieve mouse C2 (curated) CGP (chemical and genetic perturbations) gene sets.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>cgp_gene_sets =<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">&quot;Mus musculus&quot;</span>, <span class="dt">category =</span> <span class="st">&quot;C2&quot;</span>, <span class="dt">subcategory =</span> <span class="st">&quot;CGP&quot;</span>)</span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="kw">head</span>(cgp_gene_sets)</span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="co">#&gt; # A tibble: 6 x 17</span></span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="co">#&gt;   gs_cat gs_subcat gs_name entrez_gene gene_symbol human_entr… human_gene… gs_id gs_pmid</span></span>
<span id="cb6-5"><a href="#cb6-5"></a><span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;         &lt;int&gt; &lt;chr&gt;             &lt;int&gt; &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt;  </span></span>
<span id="cb6-6"><a href="#cb6-6"></a><span class="co">#&gt; 1 C2     CGP       ABBUD_…       66395 Ahnak             79026 AHNAK       M1423 145761…</span></span>
<span id="cb6-7"><a href="#cb6-7"></a><span class="co">#&gt; 2 C2     CGP       ABBUD_…       11658 Alcam               214 ALCAM       M1423 145761…</span></span>
<span id="cb6-8"><a href="#cb6-8"></a><span class="co">#&gt; 3 C2     CGP       ABBUD_…       71452 Ankrd40           91369 ANKRD40     M1423 145761…</span></span>
<span id="cb6-9"><a href="#cb6-9"></a><span class="co">#&gt; 4 C2     CGP       ABBUD_…       93760 Arid1a             8289 ARID1A      M1423 145761…</span></span>
<span id="cb6-10"><a href="#cb6-10"></a><span class="co">#&gt; 5 C2     CGP       ABBUD_…       12040 Bckdhb              594 BCKDHB      M1423 145761…</span></span>
<span id="cb6-11"><a href="#cb6-11"></a><span class="co">#&gt; 6 C2     CGP       ABBUD_…      239691 AU021092         146556 C16orf89    M1423 145761…</span></span>
<span id="cb6-12"><a href="#cb6-12"></a><span class="co">#&gt; # … with 8 more variables: gs_geoid &lt;chr&gt;, gs_exact_source &lt;chr&gt;, gs_url &lt;chr&gt;,</span></span>
<span id="cb6-13"><a href="#cb6-13"></a><span class="co">#&gt; #   gs_description &lt;chr&gt;, species_name &lt;chr&gt;, species_common_name &lt;chr&gt;,</span></span>
<span id="cb6-14"><a href="#cb6-14"></a><span class="co">#&gt; #   ortholog_sources &lt;chr&gt;, num_ortholog_sources &lt;dbl&gt;</span></span></code></pre></div>
<p>There is a helper function to show the available collections.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="kw">msigdbr_collections</span>()</span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="co">#&gt; # A tibble: 22 x 3</span></span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="co">#&gt;    gs_cat gs_subcat         num_genesets</span></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;                    &lt;int&gt;</span></span>
<span id="cb7-5"><a href="#cb7-5"></a><span class="co">#&gt;  1 C1     &quot;&quot;                         299</span></span>
<span id="cb7-6"><a href="#cb7-6"></a><span class="co">#&gt;  2 C2     &quot;CGP&quot;                     3358</span></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="co">#&gt;  3 C2     &quot;CP&quot;                        56</span></span>
<span id="cb7-8"><a href="#cb7-8"></a><span class="co">#&gt;  4 C2     &quot;CP:BIOCARTA&quot;              292</span></span>
<span id="cb7-9"><a href="#cb7-9"></a><span class="co">#&gt;  5 C2     &quot;CP:KEGG&quot;                  186</span></span>
<span id="cb7-10"><a href="#cb7-10"></a><span class="co">#&gt;  6 C2     &quot;CP:PID&quot;                   196</span></span>
<span id="cb7-11"><a href="#cb7-11"></a><span class="co">#&gt;  7 C2     &quot;CP:REACTOME&quot;             1554</span></span>
<span id="cb7-12"><a href="#cb7-12"></a><span class="co">#&gt;  8 C2     &quot;CP:WIKIPATHWAYS&quot;          587</span></span>
<span id="cb7-13"><a href="#cb7-13"></a><span class="co">#&gt;  9 C3     &quot;MIR:MIRDB&quot;               2377</span></span>
<span id="cb7-14"><a href="#cb7-14"></a><span class="co">#&gt; 10 C3     &quot;MIR:MIR_Legacy&quot;           221</span></span>
<span id="cb7-15"><a href="#cb7-15"></a><span class="co">#&gt; 11 C3     &quot;TFT:GTRD&quot;                 348</span></span>
<span id="cb7-16"><a href="#cb7-16"></a><span class="co">#&gt; 12 C3     &quot;TFT:TFT_Legacy&quot;           610</span></span>
<span id="cb7-17"><a href="#cb7-17"></a><span class="co">#&gt; 13 C4     &quot;CGN&quot;                      427</span></span>
<span id="cb7-18"><a href="#cb7-18"></a><span class="co">#&gt; 14 C4     &quot;CM&quot;                       431</span></span>
<span id="cb7-19"><a href="#cb7-19"></a><span class="co">#&gt; 15 C5     &quot;GO:BP&quot;                   7573</span></span>
<span id="cb7-20"><a href="#cb7-20"></a><span class="co">#&gt; 16 C5     &quot;GO:CC&quot;                   1001</span></span>
<span id="cb7-21"><a href="#cb7-21"></a><span class="co">#&gt; 17 C5     &quot;GO:MF&quot;                   1697</span></span>
<span id="cb7-22"><a href="#cb7-22"></a><span class="co">#&gt; 18 C5     &quot;HPO&quot;                     4494</span></span>
<span id="cb7-23"><a href="#cb7-23"></a><span class="co">#&gt; 19 C6     &quot;&quot;                         189</span></span>
<span id="cb7-24"><a href="#cb7-24"></a><span class="co">#&gt; 20 C7     &quot;&quot;                        4872</span></span>
<span id="cb7-25"><a href="#cb7-25"></a><span class="co">#&gt; 21 C8     &quot;&quot;                         302</span></span>
<span id="cb7-26"><a href="#cb7-26"></a><span class="co">#&gt; 22 H      &quot;&quot;                          50</span></span></code></pre></div>
<p>The <code>msigdbr()</code> function output is a data frame and can be manipulated using more standard methods.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>all_gene_sets <span class="op">%&gt;%</span></span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(gs_cat <span class="op">==</span><span class="st"> &quot;H&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb8-3"><a href="#cb8-3"></a><span class="st">  </span><span class="kw">head</span>()</span>
<span id="cb8-4"><a href="#cb8-4"></a><span class="co">#&gt; # A tibble: 6 x 17</span></span>
<span id="cb8-5"><a href="#cb8-5"></a><span class="co">#&gt;   gs_cat gs_subcat gs_name entrez_gene gene_symbol human_entr… human_gene… gs_id gs_pmid</span></span>
<span id="cb8-6"><a href="#cb8-6"></a><span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;         &lt;int&gt; &lt;chr&gt;             &lt;int&gt; &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt;  </span></span>
<span id="cb8-7"><a href="#cb8-7"></a><span class="co">#&gt; 1 H      &quot;&quot;        HALLMA…       11303 Abca1                19 ABCA1       M5905 &quot;&quot;     </span></span>
<span id="cb8-8"><a href="#cb8-8"></a><span class="co">#&gt; 2 H      &quot;&quot;        HALLMA…       74610 Abcb8             11194 ABCB8       M5905 &quot;&quot;     </span></span>
<span id="cb8-9"><a href="#cb8-9"></a><span class="co">#&gt; 3 H      &quot;&quot;        HALLMA…       52538 Acaa2             10449 ACAA2       M5905 &quot;&quot;     </span></span>
<span id="cb8-10"><a href="#cb8-10"></a><span class="co">#&gt; 4 H      &quot;&quot;        HALLMA…       11363 Acadl                33 ACADL       M5905 &quot;&quot;     </span></span>
<span id="cb8-11"><a href="#cb8-11"></a><span class="co">#&gt; 5 H      &quot;&quot;        HALLMA…       11364 Acadm                34 ACADM       M5905 &quot;&quot;     </span></span>
<span id="cb8-12"><a href="#cb8-12"></a><span class="co">#&gt; 6 H      &quot;&quot;        HALLMA…       11409 Acads                35 ACADS       M5905 &quot;&quot;     </span></span>
<span id="cb8-13"><a href="#cb8-13"></a><span class="co">#&gt; # … with 8 more variables: gs_geoid &lt;chr&gt;, gs_exact_source &lt;chr&gt;, gs_url &lt;chr&gt;,</span></span>
<span id="cb8-14"><a href="#cb8-14"></a><span class="co">#&gt; #   gs_description &lt;chr&gt;, species_name &lt;chr&gt;, species_common_name &lt;chr&gt;,</span></span>
<span id="cb8-15"><a href="#cb8-15"></a><span class="co">#&gt; #   ortholog_sources &lt;chr&gt;, num_ortholog_sources &lt;dbl&gt;</span></span></code></pre></div>
</div>
<div id="pathway-enrichment-analysis" class="section level2">
<h2>Pathway enrichment analysis</h2>
<p>The <code>msigdbr</code> output can be used with various popular pathway analysis packages.</p>
<p>Use the gene sets data frame for <code>clusterProfiler</code> with genes as Entrez Gene IDs.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>msigdbr_t2g =<span class="st"> </span>msigdbr_df <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(gs_name, entrez_gene) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>()</span>
<span id="cb9-2"><a href="#cb9-2"></a><span class="kw">enricher</span>(<span class="dt">gene =</span> gene_ids_vector, <span class="dt">TERM2GENE =</span> msigdbr_t2g, ...)</span></code></pre></div>
<p>Use the gene sets data frame for <code>clusterProfiler</code> with genes as gene symbols.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>msigdbr_t2g =<span class="st"> </span>msigdbr_df <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(gs_name, gene_symbol) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>()</span>
<span id="cb10-2"><a href="#cb10-2"></a><span class="kw">enricher</span>(<span class="dt">gene =</span> gene_symbols_vector, <span class="dt">TERM2GENE =</span> msigdbr_t2g, ...)</span></code></pre></div>
<p>Use the gene sets data frame for <code>fgsea</code>.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>msigdbr_list =<span class="st"> </span><span class="kw">split</span>(<span class="dt">x =</span> msigdbr_df<span class="op">$</span>gene_symbol, <span class="dt">f =</span> msigdbr_df<span class="op">$</span>gs_name)</span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="kw">fgsea</span>(<span class="dt">pathways =</span> msigdbr_list, ...)</span></code></pre></div>
<p>Use the gene sets data frame for <code>GSVA</code>.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>msigdbr_list =<span class="st"> </span><span class="kw">split</span>(<span class="dt">x =</span> msigdbr_df<span class="op">$</span>gene_symbol, <span class="dt">f =</span> msigdbr_df<span class="op">$</span>gs_name)</span>
<span id="cb12-2"><a href="#cb12-2"></a><span class="kw">gsva</span>(<span class="dt">gset.idx.list =</span> msigdbr_list, ...)</span></code></pre></div>
</div>
<div id="questions-and-concerns" class="section level2">
<h2>Questions and concerns</h2>
<p><strong>Which version of MSigDB was used?</strong></p>
<p>This package was generated with MSigDB v7.2 (released September 2020). The MSigDB version is used as the base of the package version. You can check the installed version with <code>packageVersion(&quot;msigdbr&quot;)</code>.</p>
<p><strong>Can I download the gene sets directly from MSigDB instead of using this package?</strong></p>
<p>Yes. You can then import the GMT files (with <code>getGmt()</code> from the <code>GSEABase</code> package, for example). The GMTs only include the human genes, even for gene sets generated from mouse experiments. If you are not working with non-human data, you then have to convert the MSigDB genes to your organism or your genes to human.</p>
<p><strong>Can I convert between human and mouse genes just by adjusting gene capitalization?</strong></p>
<p>That will work for most genes, but not all.</p>
<p><strong>Can I convert human genes to any organism myself instead of using this package?</strong></p>
<p>Yes. A popular method is using the <code>biomaRt</code> package. You may still end up with dozens of homologs for some genes, so additional cleanup may be helpful.</p>
<p><strong>Aren’t there already other similar tools?</strong></p>
<p>There are a few other resources that and provide some of the functionality and served as an inspiration for this package. <a href="http://ge-lab.org/#/data">Ge Lab Gene Set Files</a> has GMT files for many species. <a href="http://bioinf.wehi.edu.au/software/MSigDB/">WEHI</a> provides MSigDB gene sets in R format for human and mouse, but the genes are provided only as Entrez IDs and each collection is a separate file. <a href="https://github.com/stephenturner/msigdf">MSigDF</a> is based on the WEHI resource, but is converted to a more tidyverse-friendly data frame. When <code>msigdbr</code> was initially released, these were multiple releases behind the latest version of MSigDB, so they may not be actively maintained.</p>
</div>
<div id="details" class="section level2">
<h2>Details</h2>
<p>The Molecular Signatures Database (MSigDB) is a collection of gene sets originally created for use with the Gene Set Enrichment Analysis (GSEA) software.</p>
<p>Gene homologs are provided by HUGO Gene Nomenclature Committee at the European Bioinformatics Institute which integrates the orthology assertions predicted for human genes by eggNOG, Ensembl Compara, HGNC, HomoloGene, Inparanoid, NCBI Gene Orthology, OMA, OrthoDB, OrthoMCL, Panther, PhylomeDB, TreeFam and ZFIN. For each human equivalent within each species, only the ortholog supported by the largest number of databases is used.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
