<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1">



<title>Introduction to the msigdbr package</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Introduction to the msigdbr package</h1>



<div id="overview" class="section level2">
<h2>Overview</h2>
<p>Performing pathway analysis is a common task in genomics and there are many available software tools, many of which are R-based. Depending on the tool, it may be necessary to import the pathways into R, translate genes to the appropriate species, convert between symbols and IDs, and format the object in the required way.</p>
<p>The <code>msigdbr</code> R package provides Molecular Signatures Database (MSigDB) gene sets typically used with the Gene Set Enrichment Analysis (GSEA) software:</p>
<ul>
<li>in an R-friendly format (a data frame in a “long” format with one gene per row)</li>
<li>for multiple frequently studied model organisms (human, mouse, rat, pig, fly, yeast, etc.)</li>
<li>as both gene symbols and NCBI/Entrez Gene IDs (for better compatibility with pathway enrichment tools)</li>
<li>that can be used in a script without requiring additional external files</li>
</ul>
</div>
<div id="installation" class="section level2">
<h2>Installation</h2>
<p>The package can be installed from CRAN.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">install.packages</span>(<span class="st">&quot;msigdbr&quot;</span>)</span></code></pre></div>
</div>
<div id="usage" class="section level2">
<h2>Usage</h2>
<p>Load package.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw">library</span>(msigdbr)</span></code></pre></div>
<p>Check the available species.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="kw">msigdbr_show_species</span>()</span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="co">#&gt;  [1] &quot;Bos taurus&quot;               &quot;Caenorhabditis elegans&quot;   &quot;Canis lupus familiaris&quot;  </span></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="co">#&gt;  [4] &quot;Danio rerio&quot;              &quot;Drosophila melanogaster&quot;  &quot;Gallus gallus&quot;           </span></span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="co">#&gt;  [7] &quot;Homo sapiens&quot;             &quot;Mus musculus&quot;             &quot;Rattus norvegicus&quot;       </span></span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="co">#&gt; [10] &quot;Saccharomyces cerevisiae&quot; &quot;Sus scrofa&quot;</span></span></code></pre></div>
<p>Retrieve all human gene sets.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>m_df =<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>)</span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="kw">head</span>(m_df)</span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="co">#&gt; # A tibble: 6 x 9</span></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="co">#&gt;   gs_name     gs_id  gs_cat gs_subcat human_gene_s… species_n… entrez_g… gene_sym… sources</span></span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="co">#&gt;   &lt;chr&gt;       &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;         &lt;chr&gt;          &lt;int&gt; &lt;chr&gt;     &lt;chr&gt;  </span></span>
<span id="cb4-6"><a href="#cb4-6"></a><span class="co">#&gt; 1 AAACCAC_MI… M12609 C3     MIR       ABCC4         Homo sapi…     10257 ABCC4     &lt;NA&gt;   </span></span>
<span id="cb4-7"><a href="#cb4-7"></a><span class="co">#&gt; 2 AAACCAC_MI… M12609 C3     MIR       ABRAXAS2      Homo sapi…     23172 ABRAXAS2  &lt;NA&gt;   </span></span>
<span id="cb4-8"><a href="#cb4-8"></a><span class="co">#&gt; 3 AAACCAC_MI… M12609 C3     MIR       ACTN4         Homo sapi…        81 ACTN4     &lt;NA&gt;   </span></span>
<span id="cb4-9"><a href="#cb4-9"></a><span class="co">#&gt; 4 AAACCAC_MI… M12609 C3     MIR       ACVR1         Homo sapi…        90 ACVR1     &lt;NA&gt;   </span></span>
<span id="cb4-10"><a href="#cb4-10"></a><span class="co">#&gt; 5 AAACCAC_MI… M12609 C3     MIR       ADAM9         Homo sapi…      8754 ADAM9     &lt;NA&gt;   </span></span>
<span id="cb4-11"><a href="#cb4-11"></a><span class="co">#&gt; 6 AAACCAC_MI… M12609 C3     MIR       ADAMTS5       Homo sapi…     11096 ADAMTS5   &lt;NA&gt;</span></span></code></pre></div>
<p>Retrieve mouse hallmark collection gene sets.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>m_df =<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">&quot;Mus musculus&quot;</span>, <span class="dt">category =</span> <span class="st">&quot;H&quot;</span>)</span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="kw">head</span>(m_df)</span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="co">#&gt; # A tibble: 6 x 9</span></span>
<span id="cb5-4"><a href="#cb5-4"></a><span class="co">#&gt;   gs_name   gs_id gs_cat gs_subcat human_gen… species_… entrez_g… gene_sym… sources       </span></span>
<span id="cb5-5"><a href="#cb5-5"></a><span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;      &lt;chr&gt;         &lt;int&gt; &lt;chr&gt;     &lt;chr&gt;         </span></span>
<span id="cb5-6"><a href="#cb5-6"></a><span class="co">#&gt; 1 HALLMARK… M5905 H      &quot;&quot;        ABCA1      Mus musc…     11303 Abca1     Inparanoid,Ho…</span></span>
<span id="cb5-7"><a href="#cb5-7"></a><span class="co">#&gt; 2 HALLMARK… M5905 H      &quot;&quot;        ABCB8      Mus musc…     74610 Abcb8     Inparanoid,Ho…</span></span>
<span id="cb5-8"><a href="#cb5-8"></a><span class="co">#&gt; 3 HALLMARK… M5905 H      &quot;&quot;        ACAA2      Mus musc…     52538 Acaa2     Inparanoid,Ho…</span></span>
<span id="cb5-9"><a href="#cb5-9"></a><span class="co">#&gt; 4 HALLMARK… M5905 H      &quot;&quot;        ACADL      Mus musc…     11363 Acadl     Inparanoid,Ho…</span></span>
<span id="cb5-10"><a href="#cb5-10"></a><span class="co">#&gt; 5 HALLMARK… M5905 H      &quot;&quot;        ACADM      Mus musc…     11364 Acadm     Inparanoid,Ho…</span></span>
<span id="cb5-11"><a href="#cb5-11"></a><span class="co">#&gt; 6 HALLMARK… M5905 H      &quot;&quot;        ACADS      Mus musc…     11409 Acads     Inparanoid,Ho…</span></span></code></pre></div>
<p>Retrieve mouse C2 (curated) CGP (chemical and genetic perturbations) gene sets.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>m_df =<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">&quot;Mus musculus&quot;</span>, <span class="dt">category =</span> <span class="st">&quot;C2&quot;</span>, <span class="dt">subcategory =</span> <span class="st">&quot;CGP&quot;</span>)</span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="kw">head</span>(m_df)</span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="co">#&gt; # A tibble: 6 x 9</span></span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="co">#&gt;   gs_name   gs_id gs_cat gs_subcat human_gen… species_… entrez_g… gene_sym… sources       </span></span>
<span id="cb6-5"><a href="#cb6-5"></a><span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;      &lt;chr&gt;         &lt;int&gt; &lt;chr&gt;     &lt;chr&gt;         </span></span>
<span id="cb6-6"><a href="#cb6-6"></a><span class="co">#&gt; 1 ABBUD_LI… M1423 C2     CGP       AHNAK      Mus musc…     66395 Ahnak     Inparanoid,Ho…</span></span>
<span id="cb6-7"><a href="#cb6-7"></a><span class="co">#&gt; 2 ABBUD_LI… M1423 C2     CGP       ALCAM      Mus musc…     11658 Alcam     Inparanoid,Ho…</span></span>
<span id="cb6-8"><a href="#cb6-8"></a><span class="co">#&gt; 3 ABBUD_LI… M1423 C2     CGP       ANKRD40    Mus musc…     71452 Ankrd40   Inparanoid,Ho…</span></span>
<span id="cb6-9"><a href="#cb6-9"></a><span class="co">#&gt; 4 ABBUD_LI… M1423 C2     CGP       ARID1A     Mus musc…     93760 Arid1a    Inparanoid,Ho…</span></span>
<span id="cb6-10"><a href="#cb6-10"></a><span class="co">#&gt; 5 ABBUD_LI… M1423 C2     CGP       BCKDHB     Mus musc…     12040 Bckdhb    Inparanoid,Ho…</span></span>
<span id="cb6-11"><a href="#cb6-11"></a><span class="co">#&gt; 6 ABBUD_LI… M1423 C2     CGP       C16orf89   Mus musc…    239691 AU021092  Inparanoid,Ho…</span></span></code></pre></div>
<p>The <code>msigdbr()</code> function output can also be manipulated as a standard data frame.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>m_df =<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">&quot;Mus musculus&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(gs_cat <span class="op">==</span><span class="st"> &quot;H&quot;</span>)</span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="kw">head</span>(m_df)</span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="co">#&gt; # A tibble: 6 x 9</span></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="co">#&gt;   gs_name   gs_id gs_cat gs_subcat human_gen… species_… entrez_g… gene_sym… sources       </span></span>
<span id="cb7-5"><a href="#cb7-5"></a><span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;      &lt;chr&gt;         &lt;int&gt; &lt;chr&gt;     &lt;chr&gt;         </span></span>
<span id="cb7-6"><a href="#cb7-6"></a><span class="co">#&gt; 1 HALLMARK… M5905 H      &quot;&quot;        ABCA1      Mus musc…     11303 Abca1     Inparanoid,Ho…</span></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="co">#&gt; 2 HALLMARK… M5905 H      &quot;&quot;        ABCB8      Mus musc…     74610 Abcb8     Inparanoid,Ho…</span></span>
<span id="cb7-8"><a href="#cb7-8"></a><span class="co">#&gt; 3 HALLMARK… M5905 H      &quot;&quot;        ACAA2      Mus musc…     52538 Acaa2     Inparanoid,Ho…</span></span>
<span id="cb7-9"><a href="#cb7-9"></a><span class="co">#&gt; 4 HALLMARK… M5905 H      &quot;&quot;        ACADL      Mus musc…     11363 Acadl     Inparanoid,Ho…</span></span>
<span id="cb7-10"><a href="#cb7-10"></a><span class="co">#&gt; 5 HALLMARK… M5905 H      &quot;&quot;        ACADM      Mus musc…     11364 Acadm     Inparanoid,Ho…</span></span>
<span id="cb7-11"><a href="#cb7-11"></a><span class="co">#&gt; 6 HALLMARK… M5905 H      &quot;&quot;        ACADS      Mus musc…     11409 Acads     Inparanoid,Ho…</span></span></code></pre></div>
</div>
<div id="integrating-with-pathway-analysis-packages" class="section level2">
<h2>Integrating with Pathway Analysis Packages</h2>
<p>Use the gene sets data frame for <code>clusterProfiler</code> (for genes as Entrez Gene IDs).</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>m_t2g =<span class="st"> </span>m_df <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(gs_name, entrez_gene) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>()</span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="kw">enricher</span>(<span class="dt">gene =</span> gene_ids_vector, <span class="dt">TERM2GENE =</span> m_t2g, ...)</span></code></pre></div>
<p>Use the gene sets data frame for <code>clusterProfiler</code> (for genes as gene symbols).</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>m_t2g =<span class="st"> </span>m_df <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(gs_name, gene_symbol) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>()</span>
<span id="cb9-2"><a href="#cb9-2"></a><span class="kw">enricher</span>(<span class="dt">gene =</span> gene_symbols_vector, <span class="dt">TERM2GENE =</span> m_t2g, ...)</span></code></pre></div>
<p>Use the gene sets data frame for <code>fgsea</code>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>m_list =<span class="st"> </span>m_df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">split</span>(<span class="dt">x =</span> .<span class="op">$</span>gene_symbol, <span class="dt">f =</span> .<span class="op">$</span>gs_name)</span>
<span id="cb10-2"><a href="#cb10-2"></a><span class="kw">fgsea</span>(<span class="dt">pathways =</span> m_list, ...)</span></code></pre></div>
</div>
<div id="questions-and-concerns" class="section level2">
<h2>Questions and Concerns</h2>
<p>Which version of MSigDB was used?</p>
<blockquote>
<p>This package was generated with MSigDB v7.0 (released August 2019). The MSigDB version is used as the base of the package version. You can check the installed version with <code>packageVersion(&quot;msigdbr&quot;)</code>.</p>
</blockquote>
<p>Can I download the gene sets directly from MSigDB instead of using this package?</p>
<blockquote>
<p>Yes. You can then import the GMT files (with <code>getGmt()</code> from the <code>GSEABase</code> package, for example). The GMTs only include the human genes, even for gene sets generated from mouse data. If you are not working with human data, you then have to convert the MSigDB genes to your organism or your genes to human.</p>
</blockquote>
<p>Can I convert between human and mouse genes just by adjusting gene capitalization?</p>
<blockquote>
<p>That will work for most genes, but not all.</p>
</blockquote>
<p>Can I convert human genes to any organism myself instead of using this package?</p>
<blockquote>
<p>Yes. A popular method is using the <code>biomaRt</code> package. You may still end up with dozens of homologs for some genes, so additional cleanup may be helpful.</p>
</blockquote>
<p>Aren’t there already other similar tools?</p>
<blockquote>
<p>There are a few other resources that and provide some of the functionality and served as an inspiration for this package. <a href="http://ge-lab.org/#/data">Ge Lab Gene Set Files</a> has GMT files for many species. <a href="http://bioinf.wehi.edu.au/software/MSigDB/">WEHI</a> provides MSigDB gene sets in R format for human and mouse, but the genes are provided only as Entrez IDs and each collection is a separate file. <a href="https://github.com/stephenturner/msigdf">MSigDF</a> is based on the WEHI resource, so it provides the same data, but converted to a more tidyverse-friendly data frame. When <code>msigdbr</code> was initially released, all of them were multiple releases behind the latest version of MSigDB, so they are possibly no longer maintained.</p>
</blockquote>
</div>
<div id="details" class="section level2">
<h2>Details</h2>
<p>The Molecular Signatures Database (MSigDB) is a collection of gene sets originally created for use with the Gene Set Enrichment Analysis (GSEA) software.</p>
<p>Gene homologs are provided by HUGO Gene Nomenclature Committee at the European Bioinformatics Institute which integrates the orthology assertions predicted for human genes by eggNOG, Ensembl Compara, HGNC, HomoloGene, Inparanoid, NCBI Gene Orthology, OMA, OrthoDB, OrthoMCL, Panther, PhylomeDB, TreeFam and ZFIN. For each human equivalent within each species, only the ortholog supported by the largest number of databases is used.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
